addons/mail/static/src/new/utils/hooks.js

```diff
@@ -389,6 +389,63 @@ export function useSelection({ refName, model, preserveOnClickAwayPredicate = ()
         moveCursor(position) {
             model.start = model.end = position;
             ref.el.selectionStart = ref.el.selectionEnd = position;
+        },
+    };
+}
+
+/**
+ * @param {object} param0
+ * @param {string} [param0.refName]
+ * @param {ScrollPosition} [param0.model] Model to store saved position.
+ * @param {'bottom' | 'top'} [param0.clearOn] Whether scroll
+ * position should be cleared when reaching bottom or top.
+ */
+export function useScrollPosition({ refName, model, clearOn }) {
```

Comment:
- `refName` looks mandatory, so better as positional argument
- `model` name not provided is unconventional. Chatter is a bit special so it needs to pass model later & update it sometimes. I would also turn it positional and allow `undefined`.
- `clearOn` can also be optional positional at end